File: Data Structures & Algorithms/target-sum/submission-0.java
class Solution {
    public int findTargetSumWays(int[] nums, int target) {

    return helper(nums,  target,  0);
        
    }

    public int helper(int[] nums, int target, int i){
        if(i==nums.length){
            if(0==target){
                return 1;
            }else{
                return 0;
            }
        }
        return helper(nums,target-nums[i] ,i+1)+helper(nums,target+nums[i] ,i+1);
    }  
}

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        ans=[]
        def combine(i,curr_sum,temp):
            if curr_sum==target:
                ans.append(temp.copy())
                return 
            if i>=len(nums) or curr_sum>target:
                return 
            temp.append(nums[i])
            combine(i,curr_sum+nums[i],temp)
            temp.pop()
            combine(i+1,curr_sum,temp)
        combine(0,0,[])
        return ans
                